src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -128,28 +129,155 @@ private static bool GetCalendarInfo(string localeName, CalendarId calendarId, Ca
+        private static bool EnumDatePatterns(string localeName, CalendarId calendarId, CalendarDataType dataType, out string[] datePatterns)
+        {
+            datePatterns = null;
+
+            CallbackContext callbackContext = new CallbackContext();
+            callbackContext.DisallowDuplicates = true;
+            bool result = EnumCalendarInfo(localeName, calendarId, dataType, callbackContext);
+            if (result)
+            {
+                List<string> datePatternsList = callbackContext.Results;
+
+                datePatterns = new string[datePatternsList.Count];
+                for (int i = 0; i < datePatternsList.Count; i++)
+                {
+                    datePatterns[i] = NormalizeDatePattern(datePatternsList[i]);
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// The ICU date format characters are not exactly the same as the .NET date format characters.
+        /// NormalizeDatePattern will take in an ICU date pattern and return the equivalent .NET date pattern.
+        /// </summary>
+        /// <remarks>
+        /// see Date Field Symbol Table in http://userguide.icu-project.org/formatparse/datetime
+        /// and https://msdn.microsoft.com/en-us/library/8kb3ddd4(v=vs.110).aspx
+        /// </remarks>
+        private static string NormalizeDatePattern(string input)
+        {
+            StringBuilder destination = new StringBuilder(input.Length);
+
+            int index = 0;
+            while (index < input.Length)
+            {
+                switch (input[index])
+                {
+                    case '\'':
+                        // single quotes escape characters, like 'de' in es-SP
+                        // so read verbatim until the next single quote
+                        destination.Append(input[index++]);
+                        while (index < input.Length)
+                        {
+                            char current = input[index++];
+                            destination.Append(current);
+                            if (current == '\'')
+                            {
+                                break;
+                            }
+                        }
+                        break;
+                    case 'E':
+                    case 'e':
+                    case 'c':
+                        // 'E' in ICU is the day of the week, which maps to 3 or 4 'd's in .NET
+                        // 'e' in ICU is the local day of the week, which has no representation in .NET, but
+                        // maps closest to 3 or 4 'd's in .NET
+                        // 'c' in ICU is the stand-alone day of the week, which has no representation in .NET, but
+                        // maps closest to 3 or 4 'd's in .NET
+                        NormalizeDayOfWeek(input, destination, ref index);
+                        break;
+                    case 'L':
```

Comment:
That doesn't need to be translated, since 'M' in ICU means the same thing as 'M' in .NET.